lib/pymedphys/_dicom/connect/listen.py

```diff
@@ -0,0 +1,207 @@
+# Copyright (C) 2020 University of New South Wales & Ingham Institute
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import signal
+import sys
+import tempfile
+from pathlib import Path
+
+from pydicom.dataset import Dataset, FileDataset
+from pynetdicom import (
+    AE,
+    PYNETDICOM_IMPLEMENTATION_UID,
+    PYNETDICOM_IMPLEMENTATION_VERSION,
+    StoragePresentationContexts,
+    evt,
+)
+from pynetdicom.sop_class import (  # pylint: disable=no-name-in-module
+    VerificationSOPClass,
+)
+
+from pymedphys._dicom.connect.base import DicomConnectBase
+
+
+class DicomListener(DicomConnectBase):
+    def __init__(
+        self, storage_directory=tempfile.mkdtemp(), on_released_callback=None, **kwargs
+    ):
+        super(DicomListener, self).__init__(**kwargs)
+
+        # The directory where incoming data will be stored
+        self.storage_directory = storage_directory
+        if not isinstance(self.storage_directory, Path):
+            self.storage_directory = Path(self.storage_directory)
+
+        # Callback when association is released called with path to the directory where
+        # data from that association was stored
+        self.on_released_callback = on_released_callback
+
+        # Keep track of the directory where data for the current association is stored
+        self.association_directory = None
+
+        # The application entity
+        self.ae = None
+
+    def start(self):
+
+        # Initialise the Application Entity
+        self.ae = AE(ae_title=self.ae_title)
+
+        # Add the supported presentation context
+        self.ae.add_supported_context(VerificationSOPClass)
+        for context in StoragePresentationContexts:
+            self.ae.add_supported_context(context.abstract_syntax)
+
+        handlers = [
+            (evt.EVT_C_MOVE, self.on_c_store),
```

Comment:
C-MOVE is not C-STORE.
If the SCP receives a C-MOVE-RQ it will not be given an object to store.
Suggest deleting this for now.  
Implementing will involve being able to find objects on the filesystem, potentially only identified by SOP Instance UID (no hierarchy information), and the ability to issue a C-STORE-RQ to an entity only identified by AE Title (no IP address or port information), which then relies on the SCP having the means to get the IP address and port information based on the AE Title (e.g. via configuration or using Supplement 67 (Configuration Management)).  NB: not many implementations of Supp 67 (pity).

Reply:
You're no doubt correct, but I think I had this working in the past so that is could receive files from Orthanc which were sent via a C-MOVE from there...

I'll definitely remove it for now, and will investigate what is needed to have this working properly in the future.